src/Analyzer.JsonRuleEngine/JsonRuleResult.cs

```diff
@@ -27,8 +29,32 @@ internal class JsonRuleResult : IResult
         internal string JsonPath { get; set; }
 
         /// <summary>
-        /// Gets the expression associated with this result
+        /// Gets the expression associated with this result.
         /// </summary>
         internal Expression Expression { get; set; }
+
+        /// <summary>
+        /// Gets the actual value present at the specified path.
+        /// </summary>
+        internal JToken ActualValue { get; set; }
+
+        /// <summary>
+        /// Gets the messsage which explains why the evaluation failed.
+        /// </summary>
+        public string FailureMessage()
```

Comment:
As much as possible, I think all this logic should be in the Operators and Expressions.  I think it would be easier to understand and write if the failure messages were built right where the evaluations happen.